File: src/main/java/kanti/sl/ValueConverter.java
package kanti.sl;

import org.jetbrains.annotations.NotNull;

public interface ValueConverter {

	@NotNull
	String convert(@NotNull Object value);

	@NotNull
	Object convert(@NotNull String line);

	interface Builder {

		@NotNull
		Builder supportPrimitive(boolean support);

		@NotNull
		ValueConverter build();

	}

	@NotNull
	static Builder builder() {
		return new ValueConverterImpl.Builder();
	}

}

File: src/main/kotlin/kanti/sl/ValueConverter.kt
package kanti.sl

interface ValueConverter {

	fun convert(value: Any): String

	fun convert(line: String): Any

	interface Builder {

		fun supportPrimitive(support: Boolean): Builder

		fun build(): ValueConverter

	}

	companion object {

		@JvmStatic
		fun builder(): Builder {
			return ValueConverterImpl.Builder()
		}

	}

}